[Blog post] - Gradio’s dataframe has been upgraded! (#2769)
huggingface/blog: gradio-dataframe-upgrade.md

---
title: "Gradio’s dataframe has been upgraded!" 
thumbnail: /blog/assets/gradio-dataframe-upgrade/thumbnail.png
authors:
- user: hmb
- user: abidlabs
---

> # Gradio’s dataframe has been upgraded!

Reviewer: I wonder if we could make it catchier. What about:
```suggestion
# The NEW Gradio Dataframe component
```

Reviewer: Yeah or just: "Introducing Gradio's new Dataframe!"

> If you’ve already used Gradio’s dataframe, I think you’ll agree that it did a good job. Customisation, virtualisation, accessibility, clean UI. We've seen it used in a variety of awesome apps, like leaderboards, dashboards and interactive visualisations. We weren't complaining too much about it, though our backlog of issues had been growing, and some improvements had been in demand for a while. 

Reviewer: ```suggestion
Gradio’s `gr.Dataframe` component is one of our most popular components, we've seen it used in a variety of awesome apps, like leaderboards, dashboards, and interactive visualisations. Although we hadn't made any changes to the dataframe in quite some time, our backlog of issues had been growing, and some improvements had been in demand for a while. 
```

Well — we’re now super excited to release a host of new updates to Gradio’s dataframe component. Over the last 6 weeks, we’ve closed over 70 dataframe issues - including bugs, improvements and enhancements. 

### **1. Multi-Cell Selection**

> Finally, you can select multiple cells at once! Copy or delete values across your selection with ease.

Reviewer: Feels weird to start with "finally"
```suggestion
You can select multiple cells at once! Copy or delete values across your selection with ease.
```

</video>

### **5. Accessibility Upgrade and Enhanced Styling**

Improved keyboard navigation makes gr.Dataframe more accessible than ever. You can also take control of your dataframe’s look with a dedicated styler parameter and enhance the user experience of your app. 

<video width="600" controls autoplay loop>
  <source src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/gradio-dataframe-upgrade/a11y.mp4">
</video>

### **6. Row and Column Selection**

Access entire row data in select events for more intuitive interactivity and data manipulation.

<video width="600" controls autoplay loop>
  <source src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/gradio-dataframe-upgrade/rowcol.mp4">
</video>

### 7. Static Columns

Customise the interactivity of your dataframe by specifying non-editable columns using the `static_columns` parameter. 

<video width="600" controls autoplay loop>
  <source src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/gradio-dataframe-upgrade/static.mp4">
</video>

### **8. Search functionality**

> Quickly find the data you need with our powerful search feature by setting the `show_search` parameter to `search`. 

Reviewer: ```suggestion
Quickly find the data you need with our powerful search feature by setting the `show_search` parameter to `"search"`. 
```

<video width="600" controls autoplay loop>
  <source src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/gradio-dataframe-upgrade/filter.mp4">
</video>

### **10. Improved cell selection**

Experience smoother, more intuitive cell selection that behaves the way you expect.

<video width="600" controls autoplay loop>
  <source src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/gradio-dataframe-upgrade/cellselect.mp4">
</video>

## What’s next?

With over 70 issues closed, we’ve made huge improvements, but there’s always more for us to work on. Looking ahead, we still have more ideas to implement and we’re excited to keep refining accessibility, performance, and integration. Look out for some cool demos on our socials using the dataframe on Gradio’s [X](https://x.com/gradio).

## Try it yourself!

The updated dataframe is live in the latest Gradio release. Update your installation with `pip install --upgrade gradio`. 

<script
	type="module"
	src="https://gradio.s3-us-west-2.amazonaws.com/5.22.0/gradio.js"
></script>

> <gradio-app theme_mode="light" space="hmb/basic-dataframe"></gradio-app>

Reviewer: I upgraded this Space to use CPU-Upgrade and set the sleep amount to "never" just fyi
